lib/App/Pinto/Command/add.pm

```diff
@@ -232,6 +233,51 @@ to work on those distributions, in the event that you need to patch them
 in the future.  Be aware that most distributions do not actually declare
 their development prerequisites.
 
+=item --verify-upstream=[0-5]
+
+=item -Z [0-5]
+
+=item -Z ... -ZZZZZ
+
+!! THIS OPTION IS EXPERIMENTAL !!
+
+Require upstream distribution files to be verified before operating on them.
+Repeated use of this argument (up to 5) requires the upstream verification to
+be progressively more strict.  You can also set the verification level
+explicitly, e.g.,
+
+    --verify-upstream=3
+
+At level 0, no verification is performed. This may be useful if you need to
+override the verification level set earlier, say, in a script.
+
+At level 1, we verify the distributions checksum using the upstream CHECKSUMS
+file.  This is the default level.
+
+At level 2, we also verify the signature on the upstream CHECKSUMS file if it
+has one.  Warnings about unknown or untrusted PGP keys relating to that file
+are printed.
+
+At level 3, we also require upstream CHECKSUMS files to be signed.  Warnings
+about unknown or untrusted PGP keys relating to that file are now considered
+fatal.
+
+At level 4, we also verify the unpacked distribution using the embedded
+SIGNATURE file if it exists.  Warnings about unknown or untrusted PGP keys
+relating to that file are printed.
+
+At level 5, warnings about unknown or untrusted PGP keys relating to embedded
+SIGNATURE files are now considered fatal.
+
+Note that none of these checks are applied to LOCAL distributions, i.e.,
+distributions that do not have an upstream CHECKSUMS file.
+
+The impact of this option will largely depend on the your chosen upstream
+repositories and state of your current keyring.  Consider using a dedicated
+keyring/trustdb via the C<PINTO_GNUPGHOME> environment variable.  See the
+documentation for the L<verify|App::Pinto::Command::verify> command for the
+rationale and an example.
+
```

Comment:
Which one of these levels would be appropriate if the upstream host was a CPAN mirror.